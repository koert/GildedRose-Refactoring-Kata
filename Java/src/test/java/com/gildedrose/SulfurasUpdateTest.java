package com.gildedrose;

import org.approvaltests.combinations.CombinationApprovals;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

/**
 * @author Koert Zeilstra
 */
class SulfurasUpdateTest {

  public static final String NAME = "Sulfuras, Hand of Ragnaros";
  public static final Integer[] SELL_IN = {-1, 0, 4, 5, 6, 9, 10, 11};
  public static final Integer[] QUALITY = {0, 10, 50};

  SulfurasUpdate update = new SulfurasUpdate();

  @DisplayName("update")
  @Test
  void update() {
    CombinationApprovals.verifyAllCombinations(this::doUpdateQuality, SELL_IN, QUALITY);
  }

  String doUpdateQuality(int sellIn, int quality) {
    Item item = new Item(NAME, sellIn, quality);    
    update.update(item);
    return item.toString();
  }

}