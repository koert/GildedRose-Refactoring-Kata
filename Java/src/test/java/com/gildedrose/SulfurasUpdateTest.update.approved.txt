[-1, 0] => Sulfuras, Hand of Ragnaros, -1, 0 
[-1, 10] => Sulfuras, Hand of Ragnaros, -1, 10 
[-1, 50] => Sulfuras, Hand of Ragnaros, -1, 50 
[0, 0] => Sulfuras, Hand of Ragnaros, 0, 0 
[0, 10] => Sulfuras, Hand of Ragnaros, 0, 10 
[0, 50] => Sulfuras, Hand of Ragnaros, 0, 50 
[4, 0] => Sulfuras, Hand of Ragnaros, 4, 0 
[4, 10] => Sulfuras, Hand of Ragnaros, 4, 10 
[4, 50] => Sulfuras, Hand of Ragnaros, 4, 50 
[5, 0] => Sulfuras, Hand of Ragnaros, 5, 0 
[5, 10] => Sulfuras, Hand of Ragnaros, 5, 10 
[5, 50] => Sulfuras, Hand of Ragnaros, 5, 50 
[6, 0] => Sulfuras, Hand of Ragnaros, 6, 0 
[6, 10] => Sulfuras, Hand of Ragnaros, 6, 10 
[6, 50] => Sulfuras, Hand of Ragnaros, 6, 50 
[9, 0] => Sulfuras, Hand of Ragnaros, 9, 0 
[9, 10] => Sulfuras, Hand of Ragnaros, 9, 10 
[9, 50] => Sulfuras, Hand of Ragnaros, 9, 50 
[10, 0] => Sulfuras, Hand of Ragnaros, 10, 0 
[10, 10] => Sulfuras, Hand of Ragnaros, 10, 10 
[10, 50] => Sulfuras, Hand of Ragnaros, 10, 50 
[11, 0] => Sulfuras, Hand of Ragnaros, 11, 0 
[11, 10] => Sulfuras, Hand of Ragnaros, 11, 10 
[11, 50] => Sulfuras, Hand of Ragnaros, 11, 50 
